ccl/backupccl: TestBackupRestoreChecksum failed

[(ccl/backupccl).TestBackupRestoreChecksum failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2150004&tab=buildLog) on [master@d50061568013bce3646069c79c117abc65a49f56](https://github.com/cockroachdb/cockroach/commits/d50061568013bce3646069c79c117abc65a49f56):

```
=== RUN   TestBackupRestoreChecksum
    TestBackupRestoreChecksum: test_log_scope.go:85: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum270929600
    TestBackupRestoreChecksum: test_log_scope.go:58: use -show-logs to present logs inline
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum270929600
    TestBackupRestoreChecksum: testing.go:906: race detected during execution of test
--- FAIL: TestBackupRestoreChecksum (5.32s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestBackupRestoreChecksum PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreChecksum.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreChecksum failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2151577&tab=buildLog) on [master@92ca16fbbd8eb8378ea9bf259f5b1588e677bec8](https://github.com/cockroachdb/cockroach/commits/92ca16fbbd8eb8378ea9bf259f5b1588e677bec8):

```
=== RUN   TestBackupRestoreChecksum
    TestBackupRestoreChecksum: test_log_scope.go:85: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum568382709
    TestBackupRestoreChecksum: test_log_scope.go:58: use -show-logs to present logs inline
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum568382709
    TestBackupRestoreChecksum: testing.go:906: race detected during execution of test
--- FAIL: TestBackupRestoreChecksum (5.35s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestBackupRestoreChecksum PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreChecksum.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestBackupRestoreChecksum failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2154709&tab=buildLog) on [master@ef0e879cd95637ca0038b0406dc239d92b556c8e](https://github.com/cockroachdb/cockroach/commits/ef0e879cd95637ca0038b0406dc239d92b556c8e):

```
=== RUN   TestBackupRestoreChecksum
    TestBackupRestoreChecksum: test_log_scope.go:85: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum406730796
    TestBackupRestoreChecksum: test_log_scope.go:58: use -show-logs to present logs inline
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreChecksum406730796
    TestBackupRestoreChecksum: testing.go:906: race detected during execution of test
--- FAIL: TestBackupRestoreChecksum (5.32s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestBackupRestoreChecksum PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestBackupRestoreChecksum.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/backupccl/restore_job.go
+++ b/pkg/ccl/backupccl/restore_job.go
@@ -460,10 +460,11 @@ func restore(
 	// A note about contexts and spans in this method: the top-level context
 	// `restoreCtx` is used for orchestration logging. All operations that carry
 	// out work get their individual contexts.
+	emptyRowCount := RowCount{}
 
 	// If there weren't any spans requested, then return early.
 	if len(spans) == 0 {
-		return RowCount{}, nil
+		return emptyRowCount, nil
 	}
 
 	mu := struct {
@@ -496,7 +497,7 @@ func restore(
 	importSpans, _, err := makeImportSpans(spans, backupManifests, backupLocalityInfo,
 		highWaterMark, user, errOnMissingRange)
 	if err != nil {
-		return mu.res, errors.Wrapf(err, "making import requests for %d backups", len(backupManifests))
+		return emptyRowCount, errors.Wrapf(err, "making import requests for %d backups", len(backupManifests))
 	}
 
 	for i := range importSpans {
@@ -592,14 +593,14 @@ func restore(
 		endTime,
 		progCh,
 	); err != nil {
-		return mu.res, err
+		return emptyRowCount, err
 	}
 
 	if err := g.Wait(); err != nil {
 		// This leaves the data that did get imported in case the user wants to
 		// retry.
 		// TODO(dan): Build tooling to allow a user to restart a failed restore.
-		return mu.res, errors.Wrapf(err, "importing %d ranges", len(importSpans))
+		return emptyRowCount, errors.Wrapf(err, "importing %d ranges", len(importSpans))
 	}
 
 	return mu.res, nil

